ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+        """
+        This internal method identifies the M1/2 file number of the
+        given pedestal event in M1/2 file lists, corresponding to this run.
+
+        Parameters
+        ----------
+        pedestal_event_num: int
+            Order number of the event in the list of pedestal events
+            of the specified telescope, corresponding to this run.
+        telescope: str
+            The name of the telescope to which this event corresponds.
+            May be "M1" or "M2".
+
+        Returns
+        -------
+        file_num:
+            Order number of the corresponding file in the M1 or M2 file list.
+        """
+
+        event_id = self.pedestal_ids[telescope][pedestal_event_num]
+        file_num = np.digitize([event_id], self.event_data[telescope]['file_edges'])
+        file_num = file_num[0] - 1
+
+        return file_num
+
+    def _get_stereo_file_num(self, stereo_event_num):
+        """
+        This internal method identifies the M1/2 file numbers of the
+        given stereo event in M1/2 file lists, corresponding to this run.
+
+        Parameters
+        ----------
+        stereo_event_num: int
+            Order number of the event in the list of stereo events corresponding
+            to this run.
+
+        Returns
+        -------
+        m1_file_num:
+            Order number of the corresponding file in the M1 file list.
+        m2_file_num:
+            Order number of the corresponding file in the M2 file list.
+        """
+
+        m1_id = self.stereo_ids[stereo_event_num][0]
+        m2_id = self.stereo_ids[stereo_event_num][1]
+        m1_file_num = np.digitize([m1_id], self.event_data['M1']['file_edges'])
+        m2_file_num = np.digitize([m2_id], self.event_data['M2']['file_edges'])
+
+        m1_file_num = m1_file_num[0] - 1
+        m2_file_num = m2_file_num[0] - 1
+
+        return m1_file_num, m2_file_num
+
+    def _get_mono_file_num(self, mono_event_num, telescope):
+        """
+        This internal method identifies the M1/2 file number of the
+        given mono event in M1/2 file lists, corresponding to this run.
+
+        Parameters
+        ----------
+        mono_event_num: int
+            Order number of the event in the list of stereo events corresponding
+            to this run.
+        telescope: str
+            The name of the telescope to which this event corresponds.
+            May be "M1" or "M2".
+
+        Returns
+        -------
+        file_num:
+            Order number of the corresponding file in the M1 or M2 file list.
+        """
+
+        event_id = self.mono_ids[telescope][mono_event_num]
+        file_num = np.digitize([event_id], self.event_data[telescope]['file_edges'])
+        file_num = file_num[0] - 1
+
+        return file_num
+
+    def get_pedestal_event_data(self, pedestal_event_num, telescope):
+        """
+        This method read the photon content and arrival time (per pixel)
```

Comment:
If you like, you could remove this repetitive "this method" part from all function (or method) docstrings and try to use the "imperative" form for the first line of a docstring.

I think both PEP8 and PEP257 mention that the imperative form is preferred for the first line of a docstring, but I found this page described it also very nicely.
https://blog.dolphm.com/pep257-good-python-docstrings-by-example/

Reply:
Thanks for pointing this out. I'll be sure to follow this in future commits.